iib/workers/tasks/utils.py

```diff
@@ -280,3 +279,311 @@ def _get_function_arg_value(arg_name, func, args, kwargs):
+    }
+    if not from_index:
+        return result
+
+    with set_registry_token(overwrite_from_index_token, from_index):
+        from_index_resolved = _get_resolved_image(from_index)
+        result['arches'] = _get_image_arches(from_index_resolved)
+        result['ocp_version'] = (
+            get_image_label(from_index_resolved, 'com.redhat.index.delivery.version') or 'v4.5'
+        )
+        result['resolved_distribution_scope'] = (
+            get_image_label(from_index_resolved, 'com.redhat.index.delivery.distribution_scope')
+            or 'prod'
+        )
+        result['resolved_from_index'] = from_index_resolved
+    return result
+
+
+def get_index_image_infos(build_request_config, index_version_map):
+    """Get image infos of all images in version map."""
+    infos = {}
+    for (index, version) in index_version_map:
+        infos[index] = get_index_image_info(
+            build_request_config.overwrite_from_index_token,
+            from_index=getattr(build_request_config, index),
+            default_ocp_version=version,
+        )
+    return infos
+
+
+def gather_index_image_arches(build_request_config, index_image_infos):
+    """Gather architectures from build_request_config and provided index image."""
+    arches = set(build_request_config.add_arches or [])
+    for val in index_image_infos.values():
+        arches |= set(val['arches'])
+
+    if not arches:
+        raise IIBError('No arches were provided to build the index image')
+    return arches
+
+
+class RequestConfig:
+    """
+    Request config abstract class.
+
+    :param binary_image:  the pull specification of the container image
+                          where the opm binary gets copied from.
+    :type binary_image: str
+    :param overwrite_from_index_token: the token used for overwriting the input
+        ``from_index`` image. This is required for
+        non-privileged users to use ``overwrite_from_index``.
+        The format of the token must be
+        in the format "user:password".
+    :type overwrite_from_index_token: str
+    :param distribution_scope: the scope for distribution
+        of the index image, defaults to ``None``.
+    :type distribution_scope: str
+    :param source_from_index: the pull specification of the container image
+        containing the index that will be used
+        as a base of the merged index image.
+    :type source_from_index: str
+    :param target_index: the pull specification of the container image
+        containing the index whose new data will be added
+        to the merged index image.
+    :type target_index: str
+    :param binary_image_config: the dict of config required to
+        identify the appropriate ``binary_image`` to use.
+    :type binary_image_config: dict
+    """
+
+    _attrs = [
+        "binary_image",
+        "overwrite_from_index_token",
+        "add_arches",
+        "distribution_scope",
+        "source_from_index",
```

Comment:
If this is meant to be generic, shouldn't `source_from_index` and `target_index` not be here.